java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(Reader,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(URI);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toByteArray;(URL);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toCharArray;(InputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toCharArray;(InputStream,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toCharArray;(InputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toCharArray;(Reader);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toString;(InputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toString;(InputStream,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toString;(InputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toString;(Reader);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;toString;(URI);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toString;(URI,Charset);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toString;(URI,String);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toString;(URL);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toString;(URL,Charset);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;toString;(URL,String);;Argument[0];generated:open-url",
+        "org.apache.commons.io;IOUtils;true;write;(CharSequence,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(String,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(StringBuffer,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(byte[],Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(byte[],Writer,Charset);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(byte[],Writer,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;write;(char[],Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;writeChunked;(char[],Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;IOUtils;true;writeLines;(Collection,String,Writer);;Argument[1];generated:write-file",
```

Comment:
Strictly speaking, if we consider this a `write-file` sink, we should also consider `Argument[0].Element`, but this can't be done as of now, so I think the model generator did its job correctly :)

Something to think about for queries using `write-file` sinks though — we might need to cover this in the traditional CodeQL way (not in this PR).

Reply:
@atorralba : Thank you for all the input!!! This is really valuable!

@aschackmull : For Java model generation we explicitly declare char[] and byte[] as "bulk" types, which means that we don't generate the summary as Argument[0].Element (but instead generate Argument[0]). This is a java specialty that I left alone in the java specific part of the model generator implementation, when everything was refactored. Should we remove this Java specific part of the implementation?

Reply:
To be clear: I'm referring to `writeLines;(Collection,String,Writer);;Argument[1]`, which should also have a model for `writeLines;(Collection,String,Writer);;Argument[0].Element`,  but using access paths in sink specifications is not currently supported (afaik).

Reply:
Ah ok. That has not been generated. I will only focus on trying to fix issues related to incorrect models as a part of the update (basically my intention was just to refresh the models as the old SHA didn't work anymore). At least now we get one of the models 😄